SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1363,6 +1365,138 @@ bool RunSchemaCompiler()
+		}                                                                                                                                  \
+	} while (false)
+
+bool ExtractComponentSetFromSchemaJson(const FString& SchemaJsonPath, TMap<uint32, FComponentIDs>& OutComponentSetMap)
+{
+	TSharedPtr<FJsonValue> SchemaBundleJson;
+	{
+		TUniquePtr<FArchive> SchemaFile(IFileManager::Get().CreateFileReader(*SchemaJsonPath));
+		if (!SchemaFile)
+		{
+			return false;
+		}
+
+		TSharedRef<TJsonReader<char>> JsonReader = TJsonReader<char>::Create(SchemaFile.Get());
+
+		FJsonSerializer::Deserialize(*JsonReader, SchemaBundleJson);
+	}
+
+	const TSharedPtr<FJsonObject>* RootObject;
+
+	if (!SchemaBundleJson || !SchemaBundleJson->TryGetObject(RootObject))
+	{
+		UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Invalid schema bundle file %s"), *SchemaJsonPath);
+		return false;
+	}
+
+	const TArray<TSharedPtr<FJsonValue>>* SchemaFiles;
+	SAFE_TRYGETFIELD((*RootObject), Array, "schemaFiles", SchemaFiles);
+
+	TMap<FString, uint32> ComponentMap;
+	TMap<uint32, TSet<FString>> ComponentRefSetMap;
+
+	for (const auto& FileValue : *SchemaFiles)
+	{
+		const TSharedPtr<FJsonObject>* FileObject;
+		SAFE_TRYGET(FileValue, Object, FileObject);
+
+		const TArray<TSharedPtr<FJsonValue>>* ComponentsDecl;
+		SAFE_TRYGETFIELD((*FileObject), Array, "components", ComponentsDecl);
+
+		for (const auto& CompValue : *ComponentsDecl)
+		{
+			const TSharedPtr<FJsonObject>* CompObject;
+			if (!CompValue->TryGetObject(CompObject))
+			{
+				return false;
+			}
+
+			FString ComponentName;
+			SAFE_TRYGETFIELD((*CompObject), String, "qualifiedName", ComponentName);
+
+			int32 ComponentId;
+			SAFE_TRYGETFIELD((*CompObject), Number, "componentId", ComponentId);
+
+			ComponentMap.Add(ComponentName, ComponentId);
+		}
+
+		const TArray<TSharedPtr<FJsonValue>>* ComponentSets;
+		SAFE_TRYGETFIELD((*FileObject), Array, "componentSets", ComponentSets);
+
+		for (const auto& CompSetValue : *ComponentSets)
+		{
+			const TSharedPtr<FJsonObject>* CompSetObject;
+			SAFE_TRYGET(CompSetValue, Object, CompSetObject);
+
+			int32 ComponentSetId;
+			SAFE_TRYGETFIELD((*CompSetObject), Number, "componentSetId", ComponentSetId);
+
+			const TSharedPtr<FJsonObject>* CompListObject;
+			SAFE_TRYGETFIELD((*CompSetObject), Object, "componentList", CompListObject);
+
+			const TArray<TSharedPtr<FJsonValue>>* RefComponents;
+			SAFE_TRYGETFIELD((*CompListObject), Array, "components", RefComponents);
+
+			TSet<FString> Components;
+
+			for (const auto& CompRefValue : *RefComponents)
+			{
+				const TSharedPtr<FJsonObject>* CompRefObject;
+				SAFE_TRYGET(CompRefValue, Object, CompRefObject);
+
+				FString ComponentName;
+				SAFE_TRYGETFIELD((*CompRefObject), String, "component", ComponentName);
+
+				Components.Add(ComponentName);
+			}
+
+			ComponentRefSetMap.Add(ComponentSetId, MoveTemp(Components));
+		}
+	}
+
+	TMap<uint32, FComponentIDs> TestMap;
```

Comment:
nit: Name made me thing this was part of a test - would prefer either modifying the out map directly or naming something like `ComponentSetIdToComponentId`

Reply:
Fixed